bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/SVGUtil.java

```diff
@@ -0,0 +1,43 @@
+package org.eclipse.swt.graphics;
+
+import java.io.*;
+import java.nio.charset.*;
+
+/**
+ * Utility class for handling SVG-related operations.
+ *
+ * @since 3.129
+ */
+public class SVGUtil {
+
+	/**
+	 * Determines whether the given {@link InputStream} contains a SVG file.
+	 *
+	 * @param data byte array to check.
+	 * @return {@code true} if the input stream contains SVG content; {@code false}
+	 *         otherwise.
+	 * @throws IOException              if an error occurs while reading the stream.
+	 * @throws IllegalArgumentException if the input stream is {@code null}.
+	 */
+	public static boolean isSVGFile(byte[] data) throws IOException {
```

Comment:
I don't believe this is smth that SWT should expose as API thus it should be moved to internal package.

Reply:
SVGRasterizer needs this utility and cannot use it if it's in an internal package. Before I had the utility methods in SVGRasterizer itself but I thought it would be better to move it into SWT.

Reply:
Just imagine SWT providing PNGUtil.isPNGFile, BMPUtil.isBMPFile, GifUtil.isGifFile and so on and on. From API POV this is clear no go.

Reply:
I agree. Changed in 9648549f9e14e8cf1f8538949392162a42aa173f. I moved the utility methods back into the `JSVGRasterizer` (former `SVGRasterizer`). I also deleted the class `SVGUtil` as it was empty after the change.